garage/tf/policies/maml_policy.py

```diff
@@ -0,0 +1,153 @@
+"""Maml Policy."""
+import tensorflow as tf
+
+from garage.core import Serializable
+from garage.tf.policies.base2 import StochasticPolicy2
+
+# This is an alternative variable store for maml.
+MAML_VARIABLE_STORE = set()
+MAML_TASK_INDEX = 0
+
+
+class MamlPolicy(StochasticPolicy2):
+    """
+    MamlPolicy used by Model-Agnostic Meta-Learning.
+
+    This policy take any garage policies that use models and augment
+    them with one step adaptation tensors.
+
+    Args:
+        wrapped_policy: A garage policy using model.
+        n_tasks: Number of task for meta learning.
+        adaptation_step_size: The step size for one step adaptation.
+        name: Name of this policy.
+    """
+
+    def __init__(self,
+                 wrapped_policy,
+                 n_tasks,
+                 adaptation_step_size=0.5,
+                 name="MamlPolicy"):
+
+        self.wrapped_policy = wrapped_policy
+        self.n_tasks = n_tasks
+        self.name = name
+        self._initialized = False
+        self._adaptation_step_size = adaptation_step_size
+        self._adapted_param_store = dict()
+        self._create_update_opts()
+
+        super().__init__(wrapped_policy._env_spec)
+        Serializable.quick_init(self, locals())
+
+    def initialize(self, gradient_var, inputs):
```

Comment:
isn't this really just the `build()` method of a MamlModel?

Reply:
Yes and that was actually my previous attempt. But if we wrap a model, which is normally built during initialization, we will create many tensors (subtractions and placeholders for adaptations) that we will never use. We cannot reuse operations. The tensorboard looks kinda messy in this case. I admit that wrapping models is a better usage of models.

Reply:
perhaps you, Anson, and I can talk about it this afternoon. it sounds complicated-enough that we're unlikely to understand it over github comments.

Reply:
i think anson ended up adding an initialize API to his policies as well. it's intended to be temporary while we make it so that Policy and inherit directly from model.

this LGTM as long as you follow his _initialize convention